cmd/ocsp-updater/main.go

```diff
@@ -214,14 +216,102 @@ func (updater *OCSPUpdater) newCertificateTick(batchSize int) {
 	// OCSP responses
 	statuses, err := updater.getCertificatesWithMissingResponses(batchSize)
 	if err != nil {
+		updater.stats.Inc("OCSP.Errors.FindMissingResponses", 1, 1.0)
+		updater.log.AuditErr(fmt.Errorf("Failed to find certificates with missing OCSP responses: %s", err))
 		return
 	}
 
 	updater.generateOCSPResponses(statuses)
 }
 
+func (updater *OCSPUpdater) findRevokedCertificates(batchSize int) ([]core.CertificateStatus, error) {
+	var statuses []core.CertificateStatus
+	_, err := updater.dbMap.Select(
+		&statuses,
+		`SELECT * FROM certificateStatus
+		 WHERE status = :revoked
+		 AND ocspLastUpdated <= revokedDate
+		 LIMIT :limit`,
+		map[string]interface{}{
+			"revoked": string(core.OCSPStatusRevoked),
+			"limit":   batchSize,
+		},
+	)
+	return statuses, err
+}
+
+func (updater *OCSPUpdater) generateRevokedResponse(status core.CertificateStatus) (core.CertificateStatus, error) {
+	var cert core.Certificate
+	err := updater.dbMap.SelectOne(
+		&cert,
+		`SELECT * FROM certificates
+			 WHERE serial = :serial`,
+		map[string]interface{}{"serial": status.Serial},
+	)
+	if err != nil {
+		return core.CertificateStatus{}, err
+	}
+
+	signRequest := core.OCSPSigningRequest{
+		CertDER:   cert.DER,
+		Status:    string(core.OCSPStatusRevoked),
+		Reason:    status.RevokedReason,
+		RevokedAt: status.RevokedDate,
+	}
+
+	ocspResponse, err := updater.cac.GenerateOCSP(signRequest)
+	if err != nil {
+		return core.CertificateStatus{}, err
+	}
+
+	now := updater.clk.Now()
+	status.OCSPLastUpdated = now
+	status.OCSPResponse = ocspResponse
+	return status, nil
+}
+
+func (updater *OCSPUpdater) storeRevokedResponse(status core.CertificateStatus) error {
```

Comment:
Please move storeRevokedResponse closer to storeResponse and add detailed comments to each. At least one of the comments should explain why we are doing the `WHERE` guard.

Actually, looking more closely, can you merge them? It looks like in this function you are setting RevokedDate and RevokedReason unnecessarily. Those should already have been set by MarkCertificateRevoked, right?